test/spmd/tensor/test_view_ops.py

```diff
@@ -0,0 +1,494 @@
+        )
+        self.assertEquals(
+            view_groups([2, 3, 4, 5, 7], [12, 70]),
+            (
+                Split(
+                    Flatten(
+                        (
+                            InputDim(0),
+                            InputDim(1),
+                            InputDim(2),
+                            InputDim(3),
+                            InputDim(4),
+                        )
+                    ),
+                    (12, 70),
+                    0,
+                ),
+                Split(
+                    Flatten(
+                        (
+                            InputDim(0),
+                            InputDim(1),
+                            InputDim(2),
+                            InputDim(3),
+                            InputDim(4),
+                        )
+                    ),
+                    (12, 70),
+                    1,
+                ),
+            ),
+        )
+        self.assertEquals(
+            view_groups([2, 3, 4, 5, 7], [3, 8, 7, 5]),
+            (
+                Split(
+                    Flatten((InputDim(0), InputDim(1), InputDim(2))), (3, 8), 0
+                ),
+                Split(
+                    Flatten((InputDim(0), InputDim(1), InputDim(2))), (3, 8), 1
+                ),
+                Split(Flatten((InputDim(3), InputDim(4))), (7, 5), 0),
+                Split(Flatten((InputDim(3), InputDim(4))), (7, 5), 1),
+            ),
+        )
+        self.assertEquals(
+            view_groups([3, 4, 8, 3], [12, 4, 2, 3]),
+            (
+                Flatten((InputDim(0), InputDim(1))),
+                Split(InputDim(2), (4, 2), 0),
+                Split(InputDim(2), (4, 2), 1),
+                InputDim(3),
+            ),
+        )
+        self.assertEquals(
+            view_groups([3, 24], [1, 3, 2, 4, 1, 3, 1]),
+            (
+                Singleton(),
+                InputDim(0),
+                Split(InputDim(1), (2, 4, 3), 0),
+                Split(InputDim(1), (2, 4, 3), 1),
+                Singleton(),
+                Split(InputDim(1), (2, 4, 3), 2),
+                Singleton(),
+            ),
+        )
+        self.assertEquals(
+            view_groups([1, 1, 3, 2, 1, 1], [6, 1, 1, 1]),
+            (
+                Flatten((InputDim(2), InputDim(3))),
+                Singleton(),
+                Singleton(),
+                Singleton(),
+            ),
+        )
+        self.assertEquals(
+            view_groups([1, 1, 12, 1, 1, 1, 2, 5, 1], [3, 4, 1, 10]),
+            (
+                Split(InputDim(2), (3, 4), 0),
+                Split(InputDim(2), (3, 4), 1),
+                Singleton(),
+                Flatten((InputDim(6), InputDim(7))),
+            ),
+        )
+        self.assertEquals(
+            view_groups([2, 3, 4], [2, -1, 4]),
+            (InputDim(0), InputDim(1), InputDim(2)),
+        )
+
+    @property
+    def world_size(self) -> int:
+        return 6
```

Comment:
any reason to pick `world_size = 6`? mesh size (3, 2) seems a bit weird, is it for testing specific cases?

Reply:
Yes it's supposed to be weird because I wanted to make sure we have weird shapes to catch weird things in the tests :)